### PR TITLE
Integrate During Test

### PR DESCRIPTION
**Pull request by lamarwhitej**
Add during test to the osa_onmetal groovy script.  Setup, Start, Stop, Aggregate data.

### CODE REVIEW

File: jenkins/osa_onmetal.functions.groovy [hunk 1]
```diff
@@ -205,7 +205,51 @@ def delete_onmetal(datacenter_tag) {
 
 }
 
+def setup_during_test(host_ip) {
+    
+    sh """
+    scp -o StrictHostKeyChecking=no -r /home/ubuntu/workspace/Joshs_sandbox@3/during-upgrade-tests root@${host_ip}:/root
```
Comment (castulo) [inline jenkins/osa_onmetal.functions.groovy:211]: Why are you using this hardcoded path that is referring to a local jenkins workspace? Joshs_sandbox@3

File: jenkins/osa_onmetal.functions.groovy [hunk 2]
```diff
@@ -205,7 +205,51 @@ def delete_onmetal(datacenter_tag) {
 
 }
 
+def setup_during_test(host_ip) {
```
Comment (castulo) [inline jenkins/osa_onmetal.functions.groovy:208]: I'd prefer if this function is called something like install_during_upgrade_tests so it is consistent with the other upgrade tests suites like install_persistent_resources_tests

File: jenkins/osa_onmetal.functions.groovy [hunk 3]
```diff
@@ -205,7 +205,51 @@ def delete_onmetal(datacenter_tag) {
...
+def stop_during_test(host_ip) {
+    
+    sh """
+    ssh -o StrictHostKeyChecking=no root@${host_ip} '''
+    sudo touch /usr/during.uptime.stop
+    '''
+    """
+}
+
+def aggregate_results(host_ip) {
```
Comment (castulo) [inline jenkins/osa_onmetal.functions.groovy:239]: This function name is very vague, it does not specify what results is aggregating, is it aggregating all tests result? if that is so then it should be fine, if it only aggregates results from one test suite then a more specific name may be better

File: jenkins/osa_onmetal.functions.groovy [hunk 4]
```diff
@@ -205,7 +205,51 @@ def delete_onmetal(datacenter_tag) {
...
+
+    sh """
+    scp -o StrictHostKeyChecking=no -r root@${host_ip}:/root/output/ /home/ubuntu/
+    """
+}
+
+def parse_results() {
+  
+  sh"""
+  python elastic-benchmark/elastic_benchmark.upgrade.py
```
Comment (castulo) [inline jenkins/osa_onmetal.functions.groovy:249]: Same comment as above, does this method parses all the results or only those belonging to one test suite?

**Review by castulo — CHANGES_REQUESTED** [review]
I see your pull request is also missing the part where those functions are actually being used in the pipeline. And you need to rebase your code.

File: jenkins/osa_onmetal.functions.groovy (hunk 1 above)
Comment (dlux) [reply jenkins/osa_onmetal.functions.groovy:211]: Agree review sintaxis...

File: jenkins/osa_onmetal.functions.groovy (hunk 4 above)
Comment (lamarwhitej) [reply jenkins/osa_onmetal.functions.groovy:249]: This isn't finished, but it parse all of them and push them to elastic search.

File: jenkins/osa_onmetal.functions.groovy (hunk 3 above)
Comment (lamarwhitej) [reply jenkins/osa_onmetal.functions.groovy:239]: Yes this will pull for all of them.

File: jenkins/osa_onmetal.functions.groovy (hunk 1 above)
Comment (lamarwhitej) [reply jenkins/osa_onmetal.functions.groovy:211]: Forgot to remove from testing.

**Comment by castulo** [comment]
I'm closing this PR since I think all the comments were addressed in a separate PR.

**Pull request closed without merging**